ksql-engine/src/test/java/io/confluent/ksql/util/KafkaTopicClientTest.java

```diff
@@ -124,6 +129,58 @@ public void shouldNotFailIfTopicAlreadyExistsWhenCreating() throws InterruptedEx
     KafkaTopicClient kafkaTopicClient = new KafkaTopicClientImpl(adminClient);
     kafkaTopicClient.createTopic(topicName1, 1, (short)1, false);
     verify(adminClient);
+
+  }
+
+
+  // Compares two collections of 'kafka.clients.admin.NewTopic' objects to verify that they are
+  // the same. The NewTopic.equals method isn't implemented so it just does reference equality,
+  // which is not what we want in these tests.
+  class NewTopicComparator implements Comparator<Collection<NewTopic>> {
+
+    @Override
+    public int compare(Collection<NewTopic> topicCollection1,
+                       Collection<NewTopic> topicCollection2) {
+      if (topicCollection1.size() != topicCollection2.size()) {
+        return -1;
+      }
+
+      Iterator<NewTopic> firstIterator = topicCollection1.iterator();
+      Iterator<NewTopic> secondIterator = topicCollection2.iterator();
+
+      while (firstIterator.hasNext()) {
+        NewTopic topic1 = firstIterator.next();
+        NewTopic topic2 = secondIterator.next();
+        if (!(topic1.name().equals(topic2.name())
+              && topic1.replicationFactor() == topic2.replicationFactor()
+              && topic1.numPartitions() == topic2.numPartitions()
```

Comment:
Should these be `!=` instead of `==`?

Reply:
No. The `!` is over the whole condition. So everything has to be equal.

Reply:
Might read better as:

```
if (!topic1.name().equals(topic2.name)
    || topic1.replicationFactor() != topic2.rf()
    || etc.
```